lib/rspec/scaffold/recursive_directory_expander.rb

```diff
@@ -0,0 +1,28 @@
+module RSpec
+  module Scaffold
+    class RecursiveDirectoryExpander
+      # manages recursively finding ruby files in directories.
+
+      # @param [Pathname] input_dir
+      def initialize(input_dir)
+        @input_dir = Pathname(input_dir)
+      end
+
+      # RSpec::Scaffold::RecursiveDirectoryExpander.new("/path/to/dir")
+      def expand_ruby_files
+        # 1. Raise if not a directory
+        raise(ArgumentError.new(%Q|"#{file}" is not a directory|)) if !@input_dir.directory?
+
+        # 2. do the recursive expansion of ruby files
+        print ">> Recursively scanning #{@input_dir} for ruby files... "
```

Comment:
The name is just fancy.
`Dir.glob("#{@input_dir}**/*.rb")` takes care of looking up all ruby files in subdirecories.